From bf5549e65cacc68804be262134d8dab099b2382c Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 18 Oct 2017 17:07:38 +0300
Subject: [PATCH 2043/2241] fix: net: mvpp2x: populate of_node of net_class
 device

Patch fix DSA driver probe fail.
DSA driver failed to find netdev device by of_find_net_device_by_node
Callback, due missed of_node pointer in netdev parent device.

Fix:
Populate of_node of netdev parent device during port probe.

Change-Id: I557faf614a486b4268750920df3754b2eae50167
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45329
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index d8b2af8..16e6305 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5276,6 +5276,9 @@ skip_tso_buffers:
 	mv_pp2x_ptp_init(pdev, port, id);
 #endif
 
+	/* Populate network device of_node */
+	dev->dev.of_node = port_node;
+
 	return 0;
 	dev_err(&pdev->dev, "%s failed for port_id(%d)\n", __func__, id);
 
-- 
2.7.4

